x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/TransportDeprecationInfoAction.java

```diff
@@ -59,6 +60,14 @@ public TransportDeprecationInfoAction(Settings settings, TransportService transp
         this.indexNameExpressionResolver = indexNameExpressionResolver;
         this.settings = settings;
         this.xContentRegistry = xContentRegistry;
+        skipTheseDeprecatedSettings = DeprecationChecks.SKIP_DEPRECATIONS_SETTING.get(settings);
+        // Safe to register this here because it happens synchronously before the cluster service is started:
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(DeprecationChecks.SKIP_DEPRECATIONS_SETTING,
+            this::updateHideDeprecationsSetting);
+    }
+
+    private <T> void updateHideDeprecationsSetting(List<String> hideDeprecationsSetting) {
```

Comment:
I know this is private, but conventionally this should be `setHideDeprecationsSetting`, right? Also I might be overly paranoid, but I'd make an immutable defensive copy of the passed-in list, if I were me.

Reply:
Good point -- the method names and field names are kind of a mess after several rewrites. I thought about a defensive copy but then decided that was overkill. Now you have me questioning that...